### PR TITLE
Add New GCP / Google Workspace Related Rules

### PR DESCRIPTION
**Pull request by zestsg**
### Summary of the Pull Request

Suggesting the addition of the following rules for detecting changes to GCP/Google Workspace environments

* gcp_gworkspace_application_access_levels_modified.yml
  * This detects any modification to the applied access levels that were applied to google workspace application which could lead to a lowered security setting required to access an application.
* gcp_access_policy_deleted.yml
  * This detects any access policies that were applied to a gcp resource which could lead to a lowered security setting required to access the resource.
* gcp_breakglass_container_workload_deployed.yml
  * This detects the deployment of workloads that are deployed by using the break-glass flag to override Binary Authorization controls in GKE clusters.

### Changelog

new: GCP Break-glass Container Workload Deployed
new: Google Workspace Application Access Levels Modified
new: GCP Access Policy Deleted

### Example Log Event
N/A

### Fixed Issues
N/A

### SigmaHQ Rule Creation Conventions

- If your PR adds new rules, please consider following and applying these [conventions](https://github.com/SigmaHQ/sigma-specification/blob/main/sigmahq/sigmahq_conventions.md)

### CODE REVIEW

File: rules/cloud/gcp/audit/gcp_context_aware_access_levels_deleted.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+status: experimental
+tags:
+  - attack.persistence
+  - attack.t1098.003
+logsource:
+  product: gcp
+  service: gcp.audit
+detection:
+  selection:
+    data.protoPayload.authorizationInfo.permission: accesscontextmanager.accessLevels.delete
```
Comment (nasbench) [inline rules/cloud/gcp/audit/gcp_context_aware_access_levels_deleted.yml:20]: From the docs https://cloud.google.com/access-context-manager/docs/audit-logging this value should `accesscontextmanager.accessPolicies.accessLevels.delete` as its part of the Access Context Manager operations. But as it's covered in your rule ` GCP Access Policy Deleted` its a duplicate. So it'll be deleted.

File: rules/cloud/gcp/audit/gcp_breakglass_container_workload_deployed.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    - attack.t1548
+logsource:
+    product: gcp
+    service: gcp.audit
+detection:
+    selection:
+        data.protoPayload.resource.type: 'k8s_cluster'
+        data.protoPayload.logName:
+            - 'cloudaudit.googleapis.com/activity'
+            - 'cloudaudit.googleapis.com%2Factivity'
```
Comment (nasbench) [inline rules/cloud/gcp/audit/gcp_breakglass_container_workload_deployed.yml:21]: Added an encoded variant as the docs presented it that way and I don't have any logs to confirm this.

File: rules/cloud/gcp/gworkspace/gcp_gworkspace_application_access_levels_modified.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    - attack.privilege_escalation
+    - attack.t1098.003
+logsource:
+    product: gcp
+    service: google_workspace.admin
+detection:
+    selection:
+        eventService: 'admin.googleapis.com'
+        eventName: 'CHANGE_APPLICATION_SETTING'
+        setting_name|startswith: 'ContextAwareAccess'
```
Comment (nasbench) [inline rules/cloud/gcp/gworkspace/gcp_gworkspace_application_access_levels_modified.yml:24]: Without any logs I can't confirm that the `setting_name` **must** startswith the `ContextAwareAccess` value.

**Review by nasbench — APPROVED** [review]
Approving this solely based documentation information. As the author didn't provide necessary logs and I don't have instances to test :)

File: rules/cloud/gcp/gworkspace/gcp_gworkspace_application_access_levels_modified.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
+title: Google Workspace Application Access Levels Modified
```
Comment (nasbench) [inline rules/cloud/gcp/gworkspace/gcp_gworkspace_application_access_levels_modified.yml:1]:
```suggestion
title: Google Workspace Application Access Level Modified
```

File: rules/cloud/gcp/gworkspace/gcp_gworkspace_application_access_levels_modified.yml [hunk 5]
```diff
@@ -0,0 +1,28 @@
+title: Google Workspace Application Access Level Modified
+id: 22f2fb54-5312-435d-852f-7c74f81684ca
+status: experimental
+description: |
+    Detects when an access level is changed for a google workspace application.
```
Comment (phantinuss) [inline rules/cloud/gcp/gworkspace/gcp_gworkspace_application_access_levels_modified.yml:5]:
```suggestion
    Detects when an access level is changed for a Google workspace application.
```

File: rules/cloud/gcp/gworkspace/gcp_gworkspace_application_access_levels_modified.yml [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+    An access level is part of BeyondCorp Enterprise which is Google Workspace's way of enforcing Zero Trust model.
+    An adversary would be able to remove access levels to gain easier access to google workspace resources.
```
Comment (phantinuss) [inline rules/cloud/gcp/gworkspace/gcp_gworkspace_application_access_levels_modified.yml:7]:
```suggestion
    An adversary would be able to remove access levels to gain easier access to Google workspace resources.
```

**Review by phantinuss — APPROVED** [review]

**Pull request merged**